src/mscorlib/shared/System/String.Manipulation.cs

```diff
@@ -916,29 +916,27 @@ public string Replace(string oldValue, string newValue, bool ignoreCase, Culture
 
         public string Replace(string oldValue, string newValue, StringComparison comparisonType)
         {
+            StringSpanHelpers.CheckStringComparison(comparisonType);
+
             switch (comparisonType)
             {
                 case StringComparison.CurrentCulture:
-                    return ReplaceCore(oldValue, newValue, CultureInfo.CurrentCulture, CompareOptions.None);
-
                 case StringComparison.CurrentCultureIgnoreCase:
-                    return ReplaceCore(oldValue, newValue, CultureInfo.CurrentCulture, CompareOptions.IgnoreCase);
+                    return ReplaceCore(oldValue, newValue, CultureInfo.CurrentCulture, StringSpanHelpers.GetCaseCompareOfComparisonCulture(comparisonType));
```

Comment:
**Question:** How much value does this change add (reducing the cases in the switch statement this way)?

I have span-based APIs that have similar structure and am wondering how fruitful it would be to do the same thing there.